runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoOverlay.java

```diff
@@ -89,44 +109,174 @@ private OpponentInfoOverlay(
 		getMenuEntries().add(new OverlayMenuEntry(RUNELITE_OVERLAY_CONFIG, OPTION_CONFIGURE, "Opponent info overlay"));
 	}
 
+	public void updateRaidVars(int lastHealth)
+	{
+		if (opponentInfoConfig.mergeRaidOverlay() || opponentInfoConfig.showVanillaPercentages())
+		{
+			clientThread.invokeLater(() ->
+			{
+				Widget raidOpponentWrapper = client.getWidget(303, 6);
+				Widget raidOpponentNameWidget = client.getWidget(303, 11);
+				Widget raidOpponentHealthWidget = client.getWidget(303, 20);
+
+				if (raidOpponentNameWidget != null)
+				{
+					useVanilla = false;
+
+					if (lastHealth != 0)
+					{
+						for (String boss : opponentInfoConfig.vanillaHPBarOverride().split(","))
+						{
+							if (WildcardMatcher.matches(boss.toLowerCase().trim(), raidOpponentNameWidget.getText().toLowerCase().trim()))
+							{
+								useVanilla = true;
+								break;
+							}
+						}
+					}
+				}
+
+				if (raidOpponentWrapper != null && (!useVanilla && opponentInfoConfig.mergeRaidOverlay()))
+				{
+					raidOpponentWrapper.setHidden(true);
+					shouldUnHide = false;
+				}
+
+				if (raidOpponentNameWidget != null && raidOpponentName == null)
+				{
+					raidOpponentName = raidOpponentNameWidget.getText();
+				}
+
+				if (raidOpponentHealthWidget != null && raidOpponentMaxHealth == 0)
+				{
+					raidOpponentMaxHealth = Integer.parseInt(raidOpponentHealthWidget.getText().split(" / ")[1].split(" \\(")[0]);
+				}
+
+				else if (raidOpponentHealthWidget != null && opponentInfoConfig.showVanillaPercentages())
+				{
+					raidOpponentHealthWidget.setText(lastHealth + " / " + raidOpponentMaxHealth + " (" + String.format("%.1f", (lastHealth * 100.0) / raidOpponentMaxHealth) + "%)");
+				}
+
+				raidOpponentHealth = lastHealth;
+
+				if (lastHealth == 0)
+				{
+					//if lastHealth is zero, either the Raid Health Bar is supposed to disappear,
+					//or the opponent is dead, so remove the raid opponent name and max health vars
+					raidOpponentName = null;
+					raidOpponentMaxHealth = 0;
+
+					if (raidOpponentWrapper != null)
+					{
+						//unhide so that initial values can be read again, but wait 7 seconds so that the hp bar isn't shown
+						//while the boss is dying. It's possible to cancel the unhide by changing shouldUnHide back to false
```

Comment:
What does this mean? Is the varbit for the opponent's health not updated when the widget is hidden?

Reply:
This was the case. When the widget is hidden, the values on the widget don't update. The varbits still do update though, and that is used for the variables in the latest commit.